syntax = "proto3";

package main;


message register{
  string user_id = 1;
  string password = 2;
}

message channelPost{
  string user_id = 1;
  string channel = 2;
  string message = 3;
  string key = 4;
}

message directMessage{
  string sender_id = 1;
  string recipient = 2;
  string message = 3;
  string key = 4;
}

message login{
  string user_id = 1;
  string password = 2;
}

message watch{
  string user_id = 1;
  string channel = 2;
  string key = 3;
}

message unwatch{
  string user_id = 1;
  string channel = 2;
  string key = 3;
}

message block{
  string blocker_id = 1;
  string blocked_id = 2;
  string key = 3;
}

message unblock{
  string blocker_id = 1;
  string blocked_id = 2;
  string key = 3;
}

message response{
  string text = 1;
  bool status = 2;
}


service chatService{
  rpc registerUserOp(register) returns (response);
  rpc channelPostOp(channelPost) returns (response);
  rpc directMessageOp(directMessage) returns (response);
  rpc loginOp(login) returns (response);
  rpc watchOp(watch) returns (response);
  rpc unwatchOp(unwatch) returns (response);
  rpc blockOp(block) returns (response);
  rpc unblockOp(unblock) returns (response);
  
} 

